Google.PowerShell.Tests/Compute/ManagedGceInstanceCmdletTests.cs

```diff
@@ -148,5 +158,150 @@ public void TestRemoveGceManagedInstanceGroupPipelineZonal()
 
             instances.VerifyAll();
         }
+
+        /// <summary>
+        /// Tests that Add-GceManagedInstanceGroup works with -Region option.
+        /// </summary>
+        [Test]
+        public void TestAddGceManagedInstanceGroupByRegion()
+        {
+            string instanceGroupName = "instance-group";
+            string templateName = "instance-template";
+            Mock<RegionInstanceGroupManagersResource> instances =
+                  ServiceMock.Resource(s => s.RegionInstanceGroupManagers);
+            instances.SetupRequest(
+                  item => item.Insert(
+                      It.Is<InstanceGroupManager>(manager => manager.Name == instanceGroupName),
+                      FakeProjectId, FakeRegionName),
+                  DoneOperation);
+            instances.SetupRequest(
+                i => i.Get(FakeProjectId, FakeRegionName, instanceGroupName),
+                new InstanceGroupManager { Name = instanceGroupName });
+
+            Pipeline.Commands.AddScript($"Add-GceManagedInstanceGroup -Name {instanceGroupName} " +
+                $"-InstanceTemplate {templateName} -TargetSize 1 -Region {FakeRegionName}");
+            Collection<PSObject> results = Pipeline.Invoke();
+
+            instances.Verify(
+                resource => resource.Insert(
+                    It.Is<InstanceGroupManager>(manager => manager.Name == instanceGroupName),
+                    FakeProjectId, FakeRegionName),
+                Times.Once);
+
+            Assert.AreEqual(results.Count, 1);
+            InstanceGroupManager createdInstance = results[0].BaseObject as InstanceGroupManager;
+            Assert.AreEqual(createdInstance?.Name, instanceGroupName);
+        }
+
+        /// <summary>
+        /// Tests that Add-GceManagedInstanceGroup works with -Region and -Object.
+        /// </summary>
+        [Test]
+        public void TestAddGceManagedInstanceGroupByObjectWithRegionParam()
+        {
+            string managedRegionObject = "managedRegionObj";
+            string instanceGroupName = FirstTestGroup.Name;
```

Comment:
This test will fail if you add this line:

    FirstTestGroup.Region = $@"{ComputeHttpsLink}\region\us-central1";

If we explicitly set a target region or zone, we should not overwrite them.